optuna/trial/_trial.py

```diff
@@ -63,10 +65,40 @@ def _init_relative_params(self) -> None:
 
         study = pruners._filter_study(self.study, trial)
 
-        self.relative_search_space = self.study.sampler.infer_relative_search_space(study, trial)
-        self.relative_params = self.study.sampler.sample_relative(
-            study, trial, self.relative_search_space
-        )
+        trial_system_attrs = self.system_attrs
+        if "relative_search_space" in trial_system_attrs:
+            self._load_relative_params(study, trial, trial_system_attrs)
+        else:
+            self.relative_search_space = self.study.sampler.infer_relative_search_space(
+                study, trial
+            )
+            self.relative_params = self.study.sampler.sample_relative(
+                study, trial, self.relative_search_space
+            )
+            self._write_relative_search_space_and_params()
```

Comment:
It introduces caches for search space and parameters. This is needed to sample parameters from a joint distribution. If this mechanism doesn't exist, `optuna trial suggest` constructs `relative_params` on each time, which means that the sampling falls into independent.